packages/rocketchat-language-plugin/server/server.js

```diff
@@ -0,0 +1,46 @@
+Meteor.methods({
+	getroom() {
+		const user_id = this.userId;
+		const priority_language_fullname = Meteor.call('get_language');
+		const user_info = RocketChat.models.Users.findOneById(user_id);
+		const blacklisted = user_info.automatic_channels[0].language.blacklisted;
+		const added_to = user_info.automatic_channels[0].language.added_to;
+		const response ={
+			room_name:false,
+			create:false,
+			multiple:false,
+			invalid:false,
+			room_id:false,
+			added:added_to
+		};
+
+		if (blacklisted.includes(priority_language_fullname.name)) {
+			response.invalid = true;
+		} else {
+			if (added_to.length!==0) {
+				response.multiple = true;
+			}
+			const room = RocketChat.models.Rooms.findOneByIdOrName(priority_language_fullname.name);
+			if (room) {
+				response.room_name = room.name;
+				response.room_id = room._id;
+				const users_of_room = Meteor.call('getUsersOfRoom', room._id);
```

Comment:
This is heavy to fetch all users just to find if user is a member. There is no other function available to make a query against a database to determine this? Maybe make one. So instead of reading everything, find a way to just query a database so that computation is there.